applications/FluidDynamicsApplication/custom_processes/mass_conservation_check_process.cpp

```diff
@@ -0,0 +1,265 @@
+        "compare_to_initial_values"              : true,
+        "write_to_log_file"                      : true
+    }  )" );
+
+    rParameters.ValidateAndAssignDefaults(default_parameters);
+
+    mIsSwitchedOn = rParameters["is_switched_on"].GetBool();
+    mMassComputationFreq = rParameters["mass_computation_frequency"].GetInt();
+    mCompareToInitial = rParameters["compare_to_initial_values"].GetBool();
+    mWriteToLogFile = rParameters["write_to_log_file"].GetBool();
+}
+
+// void DistanceModificationProcess::ExecuteInitialize() {
+
+//     KRATOS_TRY;
+
+//     // Continuous distance field required variables check
+//     if (mContinuousDistance){
+//         const auto& r_node = *mrModelPart.NodesBegin();
+//         KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(NODAL_H, r_node);
+//         KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DISTANCE, r_node);
+//     }
+
+//     KRATOS_CATCH("");
+// }
+
+void MassConservationCheckProcess::ExecuteFinalizeSolutionStep()
+{
+  
+    if ( this->mIsSwitchedOn ){
+            double negVol = 0.0;
+            double posVol = 0.0;
+
+        if ( mCompareToInitial && mCounter == 1){
+            // getting initial values and storing as reference
+            this->ComputeVolumesOfFluids( posVol, negVol );
+            mInitialPositiveVolume = posVol;
+            mInitialNegativeVolume = negVol;
+        }
+
+        if ( mCounter % this->mMassComputationFreq == 0){
+            // writing an output at a given frequncy
+            this->ComputeVolumesOfFluids( posVol, negVol );
+            std::cout << " --- Volume Checking Process --- " << std::endl;
+            std::cout << "   positive Volume = " << posVol << std::endl;
+            if ( mCompareToInitial ){ 
+                std::cout << "   ( " << posVol / mInitialPositiveVolume * 100.0 << "% of initial )" << std::endl; 
+            }
+            std::cout << "   negative Volume = " << negVol << std::endl;
+            if ( mCompareToInitial ){ 
+                std::cout << "   ( " << negVol / mInitialNegativeVolume * 100.0 << "% of initial )" << std::endl; 
+            }
+            std::cout << " --- --- --- --- --- --- --- --- " << std::endl;
+
+            if (mWriteToLogFile){
+                std::ofstream myfile;
+                myfile.open ("volumeLog.txt", std::ios::out | std::ios::app);
+                if ( myfile.is_open() ){
+
+                    myfile << mCounter << "    ";
+                    myfile << posVol << "    ";
+                    if ( mCompareToInitial ){ 
+                        myfile << posVol / mInitialPositiveVolume * 100.0 << "    "; 
+                    }
+                    myfile << negVol << "    ";
+                    if ( mCompareToInitial ){ 
+                        myfile << negVol / mInitialNegativeVolume * 100.0 << "\n"; 
+                    } 
+                    myfile.close();
+                }
+            }
+        }
+
+        mCounter++;
```

Comment:
Ahh, I found that I needed [STEP] instead of [TIME_STEPS]. Now it can be replaced. Thank you for the hint. After a lot of remarks concerned this process, a lot of things were changed. It would be very kind if you could have a look at the whole new design.